### PR TITLE
Cleanup images with node

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: svor <vsvydenk@redhat.com>

- Remove node from dotnet, php, cpp, golang images;
- Update node from 10 to 12 in kubernetes-plugin image

### CODE REVIEW

**Comment by nickboldt** [comment]
Looks good! 

but why do we need node 12 in the k8s image at all?

**Comment by ericwill** [comment]
> Looks good!
> 
> but why do we need node 12 in the k8s image at all?

I looked through the history for this image and couldn't find a reason why we needed it.

**Comment by svor** [comment]
> > Looks good!
> > but why do we need node 12 in the k8s image at all?
> 
> I looked through the history for this image and couldn't find a reason why we needed it.

@nickboldt @ericwill I suspect node is needed for vscode-yaml https://github.com/svor/codeready-workspaces/blob/sv/testImages/dependencies/che-plugin-registry/v3/plugins/ms-kubernetes-tools/vscode-kubernetes-tools/1.2.0/meta.yaml#L20. 
@JPinkney can correct me if i'm wrong

**Comment by JPinkney** [comment]
Yeah, it's needed for vscode-yaml

**Comment by ericwill** [comment]
Of course it's needed for `vscode-yaml`, I even mentioned that during a standup this week :man_facepalming: sorry for the noise

File: codeready-workspaces-stacks-golang/Dockerfile [hunk 1]
```diff
@@ -72,13 +73,11 @@ RUN useradd -u 1000 -G wheel,root -d /home/jboss --shell /bin/bash -m jboss && \
 ADD entrypoint.sh ${HOME}/entrypoint.sh
 RUN mkdir -p ${HOME}/go && \
     tar -xvf /tmp/assets/codeready-workspaces-stacks-language-servers-dependencies-golang-$(uname -m).tar.gz -C /tmp && \
-    cp -R /tmp/node_modules ${HOME}/che/ls-golang && \
```
Comment (nickboldt) [inline codeready-workspaces-stacks-golang/Dockerfile:75]: So we don't need the node deps from codeready-workspaces-stacks-language-servers-dependencies-golang anymore? 

https://github.com/redhat-developer/codeready-workspaces-deprecated/blob/crw-2.6-rhel-8/golang/build.sh#L41-L45

Comment (svor) [reply codeready-workspaces-stacks-golang/Dockerfile:75]: at least I don't see any reason why we need that

Comment (nickboldt) [reply codeready-workspaces-stacks-golang/Dockerfile:75]: Asked @kcrane about this one too and he can't recall any reason either. 

So we can probably merge this safely. :D

**Review by nickboldt — APPROVED** [review]

File: codeready-workspaces-stacks-golang/Dockerfile (hunk 1 above)
Comment (nickboldt) [reply codeready-workspaces-stacks-golang/Dockerfile:75]: It's strange... seems we copy node sources for the golang lang server into the side car but then don't actually compile it. Does the lang server get built on demand somewhere?

**Pull request merged**